src/webgpu/gpu_test.ts

```diff
@@ -439,13 +448,47 @@ export class GPUTest extends Fixture {
     const typedLength = exp[0].length;
 
     const buffer = this.readSinglePixelFrom2DTexture(src, format, { x, y }, { slice, layout });
-    this.expectGPUBufferValuesPassCheck(buffer, a => checkElementsBetween(a, exp), {
+    this.expectGPUBufferValuesPassCheck(buffer, a => checkElementsBetweenFn(a, exp), {
       type: constructor,
       typedLength,
       mode: generateWarningOnly ? 'warn' : 'fail',
     });
   }
 
+  /**
+   * Equivalent to {@link expectSinglePixelBetweenTwoValuesIn2DTexture} but uses a special check func
+   * to interpret incoming values as float16
+   */
+  expectSinglePixelBetweenTwoValuesFloat16In2DTexture(
+    src: GPUTexture,
+    format: SizedTextureFormat,
+    { x, y }: { x: number; y: number },
+    {
+      exp,
+      slice = 0,
+      layout,
+      generateWarningOnly = false,
+    }: {
+      exp: [TypedArrayBufferView, TypedArrayBufferView];
```

Comment:
I think these could just be Uint16Array specifically, since anything else wouldn't make much sense.

Reply:
Tried with `Uint16Array` first but the typescript didn't compile for that (or is the vscode intellisense mark it red?)
`TypedArrayBufferView = ... | Uint16Array | ...` it seems reasonable to use Uint16Array to me. 

Any other syntax we need to make it assignable to
```
checkElementsBetweenFn?: (
        actual: TypedArrayBufferView,
        expected: readonly [TypedArrayBufferView, TypedArrayBufferView]
      ) => Error | undefined;
```
?

Reply:
Oops. I think this is for the `exp: [Uint16Array, Uint16Array]` instead. fixed

Still helps knowing why typescript wouldn't allow 
```
export function checkElementsFloat16Between(
  actual: Uint16Array,
  expected: readonly [Uint16Array, Uint16Array]
): ErrorWithExtra | undefined
```
resolves to 
```
checkElementsBetweenFn?: (
        actual: TypedArrayBufferView,
        expected: readonly [TypedArrayBufferView, TypedArrayBufferView]
      ) => Error | undefined;
```
(ErrorWithExtra => Error is not a problem)

Reply:
It's because if you take `checkElementsFloat16Between` and give it the type signature of `checkElementsBetweenFn`, then a caller to `checkElementsBetweenFn` could pass in other typed array types, which would violate the type constraints of `checkElementsFloat16Between`. This is a concept in polymorphism called "variance" (covariance / contravariance)